server/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorReport.java

```diff
@@ -30,7 +31,12 @@ public class SupervisorReport<T>
   private final DateTime generationTime;
   private final T payload;
 
-  public SupervisorReport(String id, DateTime generationTime, T payload)
+  @JsonCreator
```

Comment:
I haven't yet found the place where this deserializer is used or tested- where is that? I wanted to check something about its usage: when there's a totally-generic thing like `T` as a property, Jackson doesn't know what `T` to use unless a `TypeReference` is provided. So it's important that the deserialization be done in that specific way.

Reply:
Ah, thanks for catching this. I had intended to use this but it is not being used anymore. Will revert this.